lib/java/com/google/android/material/lists/MaterialListBinder.java

```diff
@@ -0,0 +1,25 @@
+package com.google.android.material.lists;
+
+import java.util.Arrays;
+import java.util.List;
+
+/** Interface that defines interactions between {@link MaterialViewHolder}s and data */
+public class MaterialListBinder<T, E extends MaterialViewHolder> {
+
+  protected List<T> data;
```

Comment:
The main issue is that the way we handle data is scattered. Some of it would be in the adapter and other parts are in the Recyclerview itself. Having one class to serve the purpose of handling data would be simple for developers.